### PR TITLE
Add executable scripts and refactor package structure

### PR DESCRIPTION
**Pull request by alexbarrera**
Address Issue #17 by adding a separate folder for the executable scripts. Those scripts process the command line arguments and make subsequent calls to the main methods of each package.

### CODE REVIEW

File: ipynb_gen/main.py [hunk 1]
```diff
@@ -595,61 +594,35 @@ def get_samples_by_library_type(metadata_file, sep='\t'):
...
     if required_args is None:
         required_args = ['root_dir']
 
     if optional_args is None:
         optional_args = ['user', 'sep', 'user_duke_email', 'project_name']
 
-    conf_args = {}
-    if args['conf_file']:
-        conf_args = ruamel.yaml.load(args['conf_file'], Loader=ruamel.yaml.Loader)
+    _conf_args = {}
```
Comment (tncowart) [inline ipynb_gen/main.py:604]: Why did you add all these underscores?

File: bin/ggr_cwl_ipynb_gen.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
```
Comment (tncowart) [inline bin/ggr_cwl_ipynb_gen.py:1]: should this file be named `ipynb_gen.py`?

File: ipynb_gen/consts.py [hunk 3]
```diff
@@ -47,12 +47,12 @@
 
 # Environment configuration
 CONDA_ACTIVATE = '/data/reddylab/software/miniconda2/bin/activate'
-CONTAMINATION_SCRIPT = '/data/reddylab/Darryl/GitHub/reddylab/contamination_check' # not used
+CONTAMINATION_SCRIPT = '/data/reddylab/Darryl/GitHub/reddylab/contamination_check'
```
Comment (tncowart) [inline ipynb_gen/consts.py:50]: Neither `CONTAMINATION_SCRIPT` nor `PLOT_SCRIPT` are referenced elsewhere. If they serve purposes within the context of this project I think it would be good to add comments explaining those purposes.

**Review by tncowart — CHANGES_REQUESTED** [review]
Nothing major!

File: bin/ggr_cwl_ipynb_gen.py (hunk 2 above)
Comment (alexbarrera) [reply bin/ggr_cwl_ipynb_gen.py:1]: No, naming it the same than the package broke the import. That's precisely why I changed the package name.

**Comment by tncowart** [comment]
oh, I think you can remove `.py` from the file names in `bin`.

File: ipynb_gen/consts.py (hunk 3 above)
Comment (alexbarrera) [reply ipynb_gen/consts.py:50]: The `DATA_UPLOAD_SCRIPT` was being used, but had that tag, so I took them all out. I put those two back.

File: ipynb_gen/main.py (hunk 1 above)
Comment (alexbarrera) [reply ipynb_gen/main.py:604]: I think pylint was complaining about a same name variable conflicting with this, but I moved things around and it wasn't the case. I've removed the underscores.

File: ipynb_gen/main.py [hunk 4]
```diff
@@ -200,9 +199,10 @@ def ungzip_fastq_files(conf_args, lib_type, metadata_filename=None, num_samples=
 
 def merge_fastq_files(conf_args, lib_type, metadata_filename=None, num_samples=None):
     cells = []
-    merge_fn = "%s/processing/%s/scripts/merge_lanes_%s.sh" % (conf_args['root_dir'], lib_type, conf_args['project_name'])
+    merge_fn = "%s/processing/%s/scripts/merge_lanes_%s.sh" % (
```
[the remark this answers is not in the document]
Comment (alexbarrera) [reply ipynb_gen/main.py:202]: 👍

File: ipynb_gen/main.py [hunk 5]
```diff
@@ -1,15 +1,14 @@
 #!/usr/bin/env python
 
-import argparse
 import inspect
-import sys
 import os
+import sys
+import ruamel.yaml
```
[the remark this answers is not in the document]
Comment (alexbarrera) [reply ipynb_gen/main.py:6]: Ok

**Comment by tncowart** [comment]
> oh, I think you can remove `.py` from the file names in `bin`.

Make sure the files

**Review by tncowart — APPROVED** [review]
